server/ntb/publish/ntb_ninjs.py

```diff
@@ -23,46 +17,94 @@ def __init__(self):
+            "readtime",
+            "headline",
+            "subject",
+        ]
+        for item in removable_items:
+            ninjs.pop(item)
 
         return ninjs
 
     def format_people(self, article):
         peoples = []
-        persons = article["person"]
-        for item in persons:
-            peoples.append(
-                {
-                    "name": item.get("name"),
-                    "rel": item.get("description"),
-                    "qcode": item.get("qcode"),
-                    "source": item.get("source"),
-                }
-            )
-        return peoples
-
-    def format_organisation(self, article):
+        if article.get("person"):
+            persons = article["person"]
+            for item in persons:
+                altids = item.get("altids")
+                peoples.append(
+                    {
+                        "name": item.get("name"),
+                        "rel": item.get("description"),
+                        "qcode": item.get("qcode")
+                        if altids.get("wikidata") is None
+                        else altids["wikidata"],
+                        "source": item.get("source"),
+                    }
+                )
+            return peoples
+
+    def format_organisations(self, article):
         organisations = []
-        items = article["organisation"]
-        for item in items:
-            organisations.append(
-                {
-                    "name": item.get("name"),
-                    "rel": item.get("description"),
-                    "qcode": item.get("qcode"),
-                    "source": item.get("source"),
-                }
-            )
-        return organisations
+        if article.get("organisation"):
+            items = article["organisation"]
+            for item in items:
+                altids = item.get("altids")
+                organisations.append(
+                    {
+                        "name": item.get("name"),
+                        "rel": item.get("description"),
+                        "qcode": item.get("qcode")
+                        if altids.get("wikidata") is None
+                        else altids["wikidata"],
+                        "source": item.get("source"),
+                    }
+                )
+            return organisations
+
+    def format_headlines(self, article):
+        return [{"value": article.get("headline")}]
+
+    def format_descriptions(self, ninjs):
+        return [{"value": ninjs.get("description_text")}]
+
+    def format_bodies(self, ninjs):
+        return [
+            {
+                "charcount": ninjs.get("charcount"),
+                "wordcount": ninjs.get("wordcount"),
+                "value": ninjs.get("body_html"),
+            }
+        ]
+
+    def format_subjects(self, ninjs):
+        subjects = []
+        for item in ninjs["subject"]:
+            subjects.append({"name": item["name"], "uri": item["code"]})
```

Comment:
uri should be rather `scheme:code` not just code

Reply:
so here in uri values should be mapped like `"uri" : {"scheme": item["code"]}`

Reply:
uri is a string, so it should be rather `"uri": "{scheme}:{code}".format(scheme=item["scheme"], code=item["code"])`